src/sparql/QueryObjectBuilder.ts

```diff
@@ -73,6 +79,28 @@ export default class QueryObjectBuilder {
 			},
 		];
 
+		if ( queryRepresentation.condition.propertyValueRelation === PropertyValueRelation.NotMatching ) {
```

Comment:
This method is getting quite long. But I think it is still ok and we can refactor it when we get to the multiple conditions.